examples/string-sum/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+use std::os::raw::c_char;
+use std::ptr;
+
+use pyo3_ffi::*;
+
+static mut MODULE_DEF: PyModuleDef = PyModuleDef {
+    m_base: PyModuleDef_HEAD_INIT,
+    m_name: "string_sum\0".as_ptr().cast::<c_char>(),
+    m_doc: "A Python module written in Rust.\0"
+        .as_ptr()
+        .cast::<c_char>(),
+    m_size: 0,
+    m_methods: unsafe { METHODS.as_mut_ptr().cast() },
+    m_slots: std::ptr::null_mut(),
+    m_traverse: None,
+    m_clear: None,
+    m_free: None,
+};
+
+static mut METHODS: [PyMethodDef; 2] = [
+    PyMethodDef {
+        ml_name: "sum_as_string\0".as_ptr().cast::<c_char>(),
+        ml_meth: PyMethodDefPointer {
+            _PyCFunctionFast: sum_as_string,
+        },
+        ml_flags: METH_FASTCALL,
+        ml_doc: "returns the sum of two integers as a string\0"
+            .as_ptr()
+            .cast::<c_char>(),
+    },
+    // A zeroed PyMethodDef to mark the end of the array.
+    PyMethodDef::zeroed(),
+];
+
+// The module initialization function, which must be named `PyInit_<your_module>`.
+#[allow(non_snake_case)]
+#[no_mangle]
+pub unsafe extern "C" fn PyInit_string_sum() -> *mut PyObject {
+    PyModule_Create(ptr::addr_of_mut!(MODULE_DEF))
+}
+
+pub unsafe extern "C" fn sum_as_string(
+    _self: *mut PyObject,
+    args: *mut *mut PyObject,
+    nargs: Py_ssize_t,
+) -> *mut PyObject {
+    if nargs != 2 {
+        PyErr_SetString(
+            PyExc_TypeError,
+            "sum_as_string expected 2 positional arguments\0"
+                .as_ptr()
+                .cast::<c_char>(),
+        );
+        return std::ptr::null_mut();
+    }
+
+    let (first, second) = (*args, *args.add(1));
+
+    if PyLong_Check(first) == 0 {
+        PyErr_SetString(
+            PyExc_TypeError,
+            "sum_as_string expected an int for positional argument 1\0"
+                .as_ptr()
+                .cast::<c_char>(),
+        );
+        return std::ptr::null_mut();
+    }
+
+    let first = PyLong_AsLong(first);
+    if !PyErr_Occurred().is_null() {
+        return ptr::null_mut();
```

Comment:
I know you didn't write the code, strictly speaking the error disambiguation is only needed when first is `-1`.

```suggestion
    let first = PyLong_AsLong(first);
    if first == -1 && !PyErr_Occurred().is_null() {
        return ptr::null_mut();
```